### PR TITLE
ENH: Clean up scripts

### LINKED ISSUE #115: Fix the build
It is broken. Tests from the following modules do not pass:
- typhos
- happi

See https://travis-ci.org/github/pcdshub/pcds-envs/jobs/666952497

### PR DESCRIPTION
**Pull request by ZLLentz**
A step toward addressing #115 

Fixes some minor things and tries to make the repo less "magical", e.g. avoid issues with trying to guess exactly what we want to be testing and removing more of the "manual" steps.

- adds pymongo for the happi tests (and in general)
- stop trying to guess which tests we want to run, require a list of repos to clone
- clean up the test setup implementation
- automatically update critical pins to latest on staging a new environment

Lots of packages are broken right now so I can't really do a release, I want to merge this and then go around fixing all the packages separately. A lot of stuff was depending on the old location of `SignalRO`

### CODE REVIEW

File: scripts/test_setup.py [hunk 1]
```diff
@@ -1,23 +1,62 @@
...
+
+    return version_dict
+
+
+def setup_tests(repo_file, tags=None):
+    url_base = 'https://github.com/{}.git'
+    repo_file = Path(repo_file)
+
+    with repo_file.open('r') as fd:
+        repos = fd.read().strip().split('\n')
```
Comment (klauer) [inline scripts/test_setup.py:32]: Not necessary to change, but a convenience method I've been moving to recently:
```suggestion
        repos = fd.read().strip().splitlines()
```

File: scripts/test_setup.py [hunk 2]
```diff
@@ -32,36 +71,15 @@ def pushd(new_dir):
...
+
+    if args.tag:
+        tags = version_info()
+
+        if len(tags) == 0:
+            print('No packages in current environment to test, quitting')
+    else:
+        tags = None
+
+    setup_tests(repo_file, tags=tags)
```
Comment (klauer) [inline scripts/test_setup.py:85]: Also not required, but maybe `setup_tests` becomes `setup_test` and you loop through the repos here?
```
for repo in read_repo_file_list:
   setup_test(repo, tag=tags.get(repo) if tags else None)
```

File: scripts/test_setup.py (hunk 1 above)
Comment (ZLLentz) [reply scripts/test_setup.py:32]: That's a nifty one

File: scripts/test_setup.py (hunk 2 above)
Comment (ZLLentz) [reply scripts/test_setup.py:85]: I agree with this one, I'll revise later

**Comment by ZLLentz** [comment]
Implemented Ken's suggestions, added an error message for a situation that hit me this morning while refactoring, and isolated main into a function instead of having it hang out in the if name block. Merging and moving on to the other issues on this repo.

**Pull request merged**